pkgs/development/compilers/gcc/common/platform-flags.nix

```
@@ -8,7 +8,7 @@ in lib.concatLists [
   (lib.optional (p ? cpu) "--with-cpu=${p.cpu}")
   (lib.optional (p ? abi) "--with-abi=${p.abi}")
   (lib.optional (p ? fpu) "--with-fpu=${p.fpu}")
-  (lib.optional (p ? float) "--with-float=${p.float}")
+  (lib.optional (!targetPlatform.gcc.enableMultilib or false && (p ? float)) "--with-float=${p.float}") # --with-multilib-list conflicts with --with-float
```

```suggestion
  (lib.optional (!enableMultilib && (p ? float)) "--with-float=${p.float}") # --with-multilib-list conflicts with --with-float
```
I would make `enableMultilib` another parameter so things work even if it is overridden from the target platform